File: backend/app/models/player.py
from pydantic import BaseModel


class GearItem(BaseModel):
    itemHash: int
    bucketHash: int
    name: str
    iconPath: str
    light: int


class StatEntry(BaseModel):
    name: str
    value: int


class Character(BaseModel):
    characterId: str
    classType: int
    raceType: int
    genderType: int
    light: int
    emblemBackgroundPath: str
    dateLastPlayed: str
    minutesPlayedTotal: int
    stats: dict[str, int]


class CharacterDetail(Character):
    stats: list[StatEntry]  # overrides Character.stats with resolved names
    gear: list[GearItem]
    race_name: str = ""
    race_description: str = ""
    class_name: str = ""


class ClanInfo(BaseModel):
    name: str
    callsign: str
    memberCount: int
    motto: str
    about: str
    bannerPath: str


class PlayerProfile(BaseModel):
    membershipId: str
    membershipType: int
    displayName: str
    characters: list[Character]
    clan: ClanInfo | None

File: frontend/src/domain/types/player.ts
export type ClassType = 0 | 1 | 2;
export type RaceType = 0 | 1 | 2;
export type GenderType = 0 | 1;

export interface GearItem {
  itemHash: number;
  bucketHash: number;
  name: string;
  iconPath: string;
  light: number;
}

export interface Character {
  characterId: string;
  classType: ClassType;
  raceType: RaceType;
  genderType: GenderType;
  light: number;
  emblemBackgroundPath: string;
  dateLastPlayed: string;
  minutesPlayedTotal: number;
  stats: Record<string, number>;
  class_name: string;
  race_name: string;
}

export interface CharacterDetail extends Omit<Character, 'stats'> {
  stats: Array<{ name: string; value: number }>;
  gear: GearItem[];
  race_description: string;
}

export interface ClanInfo {
  name: string;
  callsign: string;
  memberCount: number;
  motto: string;
  about: string;
  bannerPath: string;
}

export interface PlayerProfile {
  membershipId: string;
  membershipType: number;
  displayName: string;
  characters: Character[];
  clan: ClanInfo | null;
}
